lib/mayaUsd/ufe/UsdConnectionHandler.cpp

```diff
@@ -186,8 +186,37 @@ bool UsdConnectionHandler::deleteConnection(
         return false;
     }
 
-    return UsdShadeConnectableAPI::DisconnectSource(
+    bool retVal = UsdShadeConnectableAPI::DisconnectSource(
         dstUsdAttr->usdAttribute(), srcUsdAttr->usdAttribute());
+
+    // We need to make sure we cleanup on disconnection. Since having an empty connection array
```

Comment:
If the DisconnectSource return value represents success or failure, is checking connectedAttrs.empty() at line 196 sufficient, or should we check retVal as well?

Reply:
That is a very good question. We mostly assume single source connections in our code, but recent discussions on the Pixar mailing list seem to suggest they plan to use multiple connections to support array-like connection (aka the ramp shader problem).
This code tries to be a bit future looking by not assuming we will always be disconnecting a single connection. The code infrastructure and APIs to support multiple connections are already there, and it even seems the Arnold exporter is a bit ahead of the game since it uses multiple connections when exporting ramps and layered textures to Arnold.